lib/linter/apply-disable-directives.js

```diff
@@ -16,6 +16,139 @@ function compareLocations(itemA, itemB) {
     return itemA.line - itemB.line || itemA.column - itemB.column;
 }
 
+/**
+ * Groups a set of directives into sub-arrays by their parent comment.
+ * @param {Directive[]} directives Unused directives to be removed.
+ * @returns {Directive[][]} Directives grouped by their parent comment.
+ */
+function groupByParentComment(directives) {
+    const groups = [];
+
+    if (!directives.length) {
+        return groups;
+    }
+
+    let state = null;
+
+    for (const directive of directives) {
+        if (state) {
+            if (directive.unprocessedDirective.parentComment === state.parentComment) {
+                state.group.push(directive);
+            } else {
+                groups.push(state.group);
+                state = null;
+            }
+        }
+
+        state = {
+            group: [directive],
+            parentComment: directive.parentComment
+        };
+    }
+
+    if (state && state.group.length) {
+        groups.push(state.group);
+    }
+
+    return groups;
+}
+
+/**
+ * Creates removal details for a set of directives within the same comment.
+ * @param {Directive[]} directives Unused directives to be removed.
+ * @param {Token} commentToken The backing Comment token.
+ * @param {SourceCode} sourceCode A SourceCode object for the given text.
+ * @returns {{ fix, position, ruleIds }[]} Details for later creation of output Problems.
+ */
+function createIndividualDirectivesRemoval(directives, commentToken, sourceCode) {
+    const commentStart = sourceCode.lineStartIndices[commentToken.loc.start.line - 1] + commentToken.loc.start.column;
```

Comment:
```suggestion
    const commentStart = commentToken.range[0];
```

Nodes, tokens, and comments have a `range` property. We can use `commentToken.range[0]` instead of calculating it from `loc`.

Also, it looks like we don't have any tests where this function will be used?

Reply:
Nice, that `range` property -> not needing `sourceCode.lineStartIndices` makes this a lot simpler! Thanks 😄

Reply:
Awesome, that `range` property completely gets rid of the need to pass around `sourceCode`. Thanks!